core/storage/suggestion/gae_models.py

```diff
@@ -629,6 +634,33 @@ def get_in_review_question_suggestions_by_offset(
                 next_offset: int. The input offset + the number of results
                     returned by the current query.
         """
+
+        if sort_key == 'Date':
+            # The first sort property must be the same as the property to which
+            # an inequality filter is applied. Thus, the inequality filter on
+            # author_id can not be used here.
+            suggestion_query = cls.get_all().filter(
+                datastore_services.all_of(
+                    cls.status == STATUS_IN_REVIEW,
+                    cls.suggestion_type == feconf.SUGGESTION_TYPE_ADD_QUESTION,
+            )).order(-cls.created_on)
+
+            sorted_results: List[GeneralSuggestionModel] = []
+
+            while len(sorted_results) < limit:
+                suggestion_model: Sequence[GeneralSuggestionModel] = (
+                    suggestion_query.fetch(1, offset=offset))
```

Comment:
Fetching one by one will take a very long time. Why not fetch in larger batches (say 1000), go through the results, and filter manually as needed?

Please test this locally with 10000 items, try to make it so that the difference between 10k items and 1 item is not high (you can achieve this by doing a single fetch call rather than many).

Reply:
That's a good suggestion.

I tested the new codes locally. The execution time of the function `get_in_review_question_suggestions_by_offset()` is around 0.03s for 10k items, and around 0.002s for 1 item.